### PR TITLE
[Feature] Hard Line Break Support

### PR DESCRIPTION
**Pull request by KSR-Yasuda**
Fix for #163.

Introduces `g:previm_hard_line_break` to switch soft / hard line break mode.

### CODE REVIEW

**Comment by KSR-Yasuda** [comment]
Also introduced buffer-wise option `b:previm_hard_line_break`.

File: autoload/previm.vim [hunk 1]
```diff
@@ -137,6 +137,10 @@ function! s:function_template() abort
       \ printf('return %s;', get(g:, 'previm_show_header', 1)),
       \ '}',
       \ '',
+      \ 'function isHardLineBreak() {',
+      \ printf('return %s;', exists('b:previm_hard_line_break') ? (get(b:, 'previm_hard_line_break', v:false) ? 'true' : 'false') : (get(g:, 'previm_hard_line_break', v:false) ? 'true' : 'false')),
```
Comment (mattn) [inline autoload/previm.vim:141]: No need to check with exists if you use `get(b:, ...)`. Right?

Comment (mattn) [inline autoload/previm.vim:141]:
```suggestion
      \ printf('return %s;', get(b:, 'previm_hard_line_break', v:false) ? 'true' : 'false'),
```

**Comment by mattn** [comment]
Merged https://github.com/previm/previm/commit/a65729d10c28f4aff1aacd4d0685e9ffd088c106

Thanks!

**Comment by KSR-Yasuda** [comment]
So, finally buffer-wise setting `b:previm_hard_line_break` is discarded?

I thought goes better to temporarily set hard-line-break feature.

For example,
you live with soft-line-break setting (`g:previm_hard_line_break = 0`),
but you want to take a glance at a hard-line-break style document.

If you look hard- and soft-line-break docs alternatively,
I though setting buffer-wise line-break style is convenient:
because line-break style is file associated.

**Comment by KSR-Yasuda** [comment]
And now it has a bug: always works as hard-line-break style.

It looks you have reworked around option settings,
but probably it should be as below:

```diff
--- a/autoload/previm.vim
+++ b/autoload/previm.vim
@@ -408,7 +408,7 @@ function! previm#options()
   endif
   return json_encode({
   \   'imagePrefix': get(g:, 'previm_plantuml_imageprefix', v:null),
-  \   'hardLineBreak': get(g:, 'previm_hard_line_break', v:false) ? 'true' : 'false',
+  \   'hardLineBreak': get(g:, 'previm_hard_line_break', v:false),
   \   'showheader': get(g:, 'previm_show_header', 1),
   \   'autoClose': get(g:, 'previm_auto_close', 0),
   \ })
```

----

In my first proposal, it was the code to print JavaScript code segment.

Because it was a `printf()` argument, I changed the setting into a string of (`true` or `false`).

```diff
--- a/autoload/previm.vim
+++ b/autoload/previm.vim
@@ -137,6 +137,10 @@ function! s:function_template() abort
       \ printf('return %s;', get(g:, 'previm_show_header', 1)),
       \ '}',
       \ '',
+      \ 'function isHardLineBreak() {',
+      \ printf('return %s;', exists('b:previm_hard_line_break') ? (get(b:, 'previm_hard_line_break', v:false) ? 'true' : 'false') : (get(g:, 'previm_hard_line_break', v:false) ? 'true' : 'false')),
+      \ '}',
+      \ '',
       \ 'function getFileName() {',
       \ printf('return "%s";', s:escape_backslash(current_file)),
       \ '}',
```

In your reworked code, it looks in Vim layer.

So if you convert the option into a string, it always evaluated as true.

**Comment by KSR-Yasuda** [comment]
If you have a choice to re-enable buffer-wise option, fix as below:

```diff
diff --git a/autoload/previm.vim b/autoload/previm.vim
index 4892961..bc60c06 100644
--- a/autoload/previm.vim
+++ b/autoload/previm.vim
@@ -413,7 +413,7 @@ function! previm#options()
   endif
   return json_encode({
   \   'imagePrefix': get(g:, 'previm_plantuml_imageprefix', v:null),
-  \   'hardLineBreak': get(g:, 'previm_hard_line_break', v:false) ? 'true' : 'false',
+  \   'hardLineBreak': get(b:, 'previm_hard_line_break', get(g:, 'previm_hard_line_break', v:false)),
   \   'showheader': get(g:, 'previm_show_header', 1),
   \   'autoClose': get(g:, 'previm_auto_close', 0),
   \ })
```

**Pull request closed without merging**